onnxruntime/core/providers/cpu/tensor/slice.cc

```diff
@@ -304,22 +333,40 @@ static Status SliceImpl(OpKernelContext* ctx,
     flattened_input_dims.back() = compute_metadata.p_flattened_output_dims_->back();
     TensorShape input_shape(std::move(flattened_input_dims));
 
-    auto input_iterator = SliceIterator<T>(input_tensor, input_shape, compute_metadata.starts_, *compute_metadata.p_flattened_output_dims_, compute_metadata.steps_);
+    auto input_iterator = SliceIterator<T>(input_tensor, input_shape, compute_metadata.starts_,
+                                           *compute_metadata.p_flattened_output_dims_, compute_metadata.steps_);
     create_output(input_iterator);
   } else {
-    auto input_iterator = SliceIterator<T>(input_tensor, compute_metadata.starts_, compute_metadata.output_dims_, compute_metadata.steps_);
+    auto input_iterator = SliceIterator<T>(input_tensor, compute_metadata.starts_, compute_metadata.output_dims_,
+                                           compute_metadata.steps_);
     create_output(input_iterator);
   }
 
   return Status::OK();
 }
 
+template <typename EnabledTypes, typename T>
+static inline bool CallSliceImplIfEnabled(OpKernelContext* ctx,
```

Comment:
>CallSliceImplIfEnabled [](start = 19, length = 22)

wonder if this can be extended to call arbitrary enable-able code with a lambda...

Reply:
generic lambdas were c++14 and we're currently stuck at c++11, so I didn't think doing it as a lambda was possible.

---
In reply to: [569695618](https://github.com/microsoft/onnxruntime/pull/6547#discussion_r569695618) [](ancestors = 569695618)